src/utils/transaction.ts

```diff
@@ -8,10 +8,74 @@
 import { Utxo } from '../wallet/types';
 import { UtxoError } from '../errors';
 import { HistoryTransactionOutput } from '../models/types';
-import wallet from '../wallet';
-import {crypto as cryptoBL, PrivateKey} from 'bitcore-lib'
+import {crypto as cryptoBL, PrivateKey, HDPrivateKey} from 'bitcore-lib'
+import { TOKEN_AUTHORITY_MASK, TOKEN_MINT_MASK, TOKEN_MELT_MASK, HATHOR_TOKEN_CONFIG, CREATE_TOKEN_TX_VERSION, DEFAULT_TX_VERSION } from '../constants';
+import Transaction from '../models/transaction';
+import CreateTokenTransaction from '../models/create_token_transaction';
+import Input from '../models/input';
+import Output from '../models/output';
+import Network from '../models/network';
+import { IBalance, IStorage, IHistoryTx, IDataOutput, IDataTx, isDataOutputCreateToken, IHistoryOutput, IDataInput } from '../types';
+import Address from '../models/address';
+import P2PKH from '../models/p2pkh';
+import P2SH from '../models/p2sh';
+import ScriptData from '../models/script_data';
+import { ParseError } from '../errors';
+import helpers from './helpers';
+import { getAddressType } from './address';
 
 const transaction = {
+
+  /**
+   * Check if the output is an authority output
+   *
+   * @param {Pick<HistoryTransactionOutput, 'token_data'>} output An output with the token_data field
+   * @returns {boolean} If the output is an authority output
+   */
+  isAuthorityOutput(output: Pick<HistoryTransactionOutput, 'token_data'>): boolean {
+    return (output.token_data & TOKEN_AUTHORITY_MASK) > 0;
+  },
+
+  /**
+   * Check if the output is a mint authority output
+   *
+   * @param {Pick<HistoryTransactionOutput, 'token_data'|'value'>} output An output with the token_data and value fields
+   * @returns {boolean} If the output is a mint authority output
+   */
+  isMint(output: Pick<HistoryTransactionOutput, 'token_data' | 'value'>): boolean {
+    return this.isAuthorityOutput(output) && ((output.value & TOKEN_MINT_MASK) > 0);
+  },
+
+  /**
+   * Check if the output is a melt authority output
+   *
+   * @param {Pick<HistoryTransactionOutput, 'token_data'|'value'>} output An output with the token_data and value fields
+   * @returns {boolean} If the output is a melt authority output
+   */
+  isMelt(output: Pick<HistoryTransactionOutput, 'token_data' | 'value'>): boolean {
+    return this.isAuthorityOutput(output) && ((output.value & TOKEN_MELT_MASK) > 0);
+  },
+
+  /**
+   * Check if the utxo is locked
+   *
+   * @param {Pick<HistoryTransactionOutput, 'decoded'>} output The output to check
+   * @param {{refTs: number|undefined}} options Use these values as reference to check if the output is locked
+   * @returns {boolean} Wheather the output is locked or not
+   */
+  isOutputLocked(
+    output: Pick<HistoryTransactionOutput, 'decoded'>,
+    options: { refTs?: number } = {},
+  ): boolean {
+    // XXX: check reward lock: requires blockHeight, bestBlockHeight and reward_spend_min_blocks
```

Comment:
It checks timelock only, but reward lock (or height lock) requires knowledge of the current network height, how many blocks are required to spend a block reward and the height of the first block which confirms the transactions this utxo belongs to.

Since we require many other outside data to verify reward lock it would be best to check this in some other place that has these information.